static/js/Misc.jsx

```diff
@@ -1105,94 +1106,100 @@ function useHiddenButtons() {
+  `type` is 'sources', 'cats', 'books' or 'topics'
+  `data` is list when `type` === 'cats' which tells us where we are in the TOC tree,
+        for `type` === 'books' it's the name of the book
         for `type` === 'topics' it's a dictionary of the topic object
         for `type` === 'sources' it's a list where the first item is topic slug and second item is source data
    */
   const [editCategory, toggleEditCategory] = useEditToggle();
   const [addCategory, toggleAddCategory] = useEditToggle();
   const [reorderCategory, toggleReorderCategory] = useEditToggle();
   const [addSource, toggleAddSource] = useEditToggle();
+  const [addSection, toggleAddSection] = useEditToggle();
   const [hiddenButtons, setHiddenButtons] = useHiddenButtons(true);
 
   const adminClasses = classNames({adminButtons: 1, hiddenButtons});
   let adminButtonsSpan = null;
   if (Sefaria.is_moderator && editCategory) {
-    if (data.length === 0) {  // at /texts or /topics
-      const url = type === 'topics' ? `/api/topic/reorder` : `/api/category?reorder=1`;
-      const redirect = type === 'topics' ? '/topics' : '/texts';
-      const origItems = type === 'topics' ? Sefaria.topic_toc : Sefaria.toc;
-      adminButtonsSpan = <ReorderEditor close={toggleEditCategory} type={type} origItems={origItems}
-                                        postURL={url} redirect={redirect}/>;
-    } else if (type === "sources") {
-      const [topicSlug, refData] = data;
-      adminButtonsSpan = <SourceEditor topic={topicSlug} origData={refData} close={toggleEditCategory}/>;
-    } else if (type === "books") {
-      let tocObject = Sefaria.tocObjectByCategories(data);
-      const origDesc = {en: tocObject.enDesc, he: tocObject.heDesc};
-      const origCategoryDesc = {en: tocObject.enShortDesc, he: tocObject.heShortDesc};
-      const origData = {
-        origEn: tocObject.category,
-        origHe: tocObject.heCategory,
-        origDesc,
-        origCategoryDesc,
-        isPrimary: tocObject.isPrimary
-      };
-      adminButtonsSpan = <CategoryEditor origData={origData} close={toggleEditCategory} origPath={data.slice(0, -1)}/>;
-    } else if (type === "topics") {
-      const initCatSlug = TopicToCategorySlug(data);
-      const origData = {
-        origSlug: data.slug, origCategorySlug: initCatSlug,
-        origEn: data.primaryTitle.en, origHe: data.primaryTitle.he || ""
-      };
-      origData.origDesc = data.description || {"en": "", "he": ""};
-      origData.origCategoryDesc = data.categoryDescription || {"en": "", "he": ""};
-      const origWasCat = "displays-above" in data?.links;
-      adminButtonsSpan = <TopicEditor origData={origData}
-                                      origWasCat={origWasCat}
-                                      onCreateSuccess={(slug) => window.location.href = "/topics/" + slug}
-                                      close={toggleEditCategory}/>;
-    }
+      if (data.length === 0) {  // at /texts or /topics
```

Comment:
This code is very long and has many nested conditions, try simplifying it by breaking it down into subcomponents that can be more maintainable and readable:

```const CategoryHeader = ({children, type, data = [], edit = true, add_subcategory = true, reorder = false, add_source = false, add_section = false}) => {
  // Keep existing hooks as they are
  const [editCategory, toggleEditCategory] = useEditToggle();
  const [addCategory, toggleAddCategory] = useEditToggle();
  const [reorderCategory, toggleReorderCategory] = useEditToggle();
  const [addSource, toggleAddSource] = useEditToggle();
  const [addSection, toggleAddSection] = useEditToggle();
  const [hiddenButtons, setHiddenButtons] = useHiddenButtons(true);
  
  // Rest of the code ...

  const getAdminButtons = () => {
    if (Sefaria.is_moderator && editCategory) {
        return getAdminButtonForEditCategory();
    } else if (Sefaria.is_moderator && addSource && type === "topics") {
        return <SourceEditor topic={data.slug} close={toggleAddSource}/>;
    } else if (Sefaria.is_moderator && addCategory) {
        return getAdminButtonForAddCategory();
    } else if (Sefaria.is_moderator && addSection && type === 'books') {
        window.location = `/add/${data}`;
    } else if (Sefaria.is_moderator && reorderCategory) {
        return getAdminButtonForReorderCategory();
    } else if (Sefaria.is_moderator) {
        return getAdminButtonDefault();
    }
  }

  // Then you create each of these functions separately, here's an example for one
  const getAdminButtonForEditCategory = () => {
    // Move the logic for the editCategory case here, and return the appropriate component
    // Here's an example, but you would add the rest of the logic as well
    if (data.length === 0) {  // at /texts or /topics
      const url = type === 'topics' ? `/api/topic/reorder` : `/api/category?reorder=1`;
      const redirect = type === 'topics' ? '/topics' : '/texts';
      const origItems = type === 'topics' ? Sefaria.topic_toc : Sefaria.toc;
      return <ReorderEditor close={toggleEditCategory} type={type} origItems={origItems}
                            postURL={url} redirect={redirect}/>;
    } 
    // Add else if blocks for the other conditions here ...
  }

  // Create the other functions (getAdminButtonForAddCategory, getAdminButtonForReorderCategory, etc.) in a similar way

  // Now getAdminButtons function is much simpler and easier to read

  const adminButtonsSpan = getAdminButtons();

  // Rest of the code ...

  return <span className={wrapper}><span onMouseEnter={() => setHiddenButtons()}>{children}</span><span>{adminButtonsSpan}</span></span>;
}
```